twill/src/main/java/org/apache/accumulo/testing/twill/TestRunner.java

```
@@ -0,0 +1,166 @@
+    List<String> mainArgs = new ArrayList<>();
+  }
+
+  private static void verifyPath(String path) {
+    File f = new File(path);
+    Preconditions.checkState(f.exists());
+    Preconditions.checkState(f.canRead());
+  }
+
+  public static void main(String[] args) throws Exception {
+
+    TestRunnerOpts opts = new TestRunnerOpts();
+    new JCommander(opts, args);
+
+    verifyPath(opts.jarPath);
+    verifyPath(opts.testProps);
+    verifyPath(opts.logProps);
+
+    String[] mainArgs = opts.mainArgs.stream().toArray(String[]::new);
+    BundledJarRunner.Arguments arguments = new BundledJarRunner.Arguments(opts.jarPath, "/lib",
+                                                                          opts.mainClass, mainArgs);
+
+    Properties props = new Properties();
+    FileInputStream fis = new FileInputStream(opts.testProps);
+    props.load(fis);
+    fis.close();
+    String zookeepers = props.getProperty(TestProps.ZOOKEEPERS);
+
+    final TwillRunnerService twillRunner = new YarnTwillRunnerService(new YarnConfiguration(),
+                                                                      zookeepers);
+    twillRunner.start();
+
+    final TwillController controller = twillRunner.prepare(
+        new TestRunnerApp(opts, props))
+        .addJVMOptions("-Dlog4j.configuration=file:$PWD/" + new File(opts.logProps).getName())
+        .withArguments("BundledJarRunnable", arguments.toArray())
+        .start();
+
+    final AtomicBoolean done = new AtomicBoolean(false);
+
+    Runtime.getRuntime().addShutdownHook(new Thread(() -> {
+      try {
+        if (!done.get()) {
+          controller.kill();
+        }
+      } finally {
+        twillRunner.stop();
+      }
+    }));
+
+    LOG.info("Waiting for {} to finish in YARN...", opts.testName);
+    LOG.info("Press ctrl-c to kill {} in YARN", opts.testName);
+
+    try {
+      controller.awaitTerminated();
+      done.set(true);
+    } catch (ExecutionException e) {
+      LOG.error("Error", e);
```

Wouldn't we want to re-throw the exception so that we get a non-zero exit code?